datalad/plugin/add_readme.py

```diff
@@ -56,11 +50,51 @@ def dlplugin(dataset, filename='README.rst', existing='skip'):
         return
 
     # unlock, file could be annexed
-    # TODO yield
     if lexists(filename):
         dataset.unlock(filename)
 
-    with open(filename, 'a' if existing == 'append' else 'w') as fp:
+    # get any metadata on the dataset itself
+    dsinfo = dataset.metadata('.', reporton='datasets', return_type='item-or-list')
+    if not isinstance(dsinfo, dict) or dsinfo.get('status', None) != 'ok':
+        lgr.warn("Could not obtain dataset metadata, proceeding without")
+        dsinfo = {}
+        meta = {}
+    else:
+        meta = dsinfo['metadata']
+
+    metainfo = ''
+    for label, content in (
+            ('', meta.get('description', meta.get('shortdescription', ''))),
+            ('Author{}'.format('s' if isinstance(meta.get('author', None), list) else ''),
+                u'\n'.join([u'- {}'.format(a) for a in assure_list(meta.get('author', []))])),
+            ('Homepage', meta.get('homepage', '')),
+            ('Reference', meta.get('citation', '')),
+            ('License', meta.get('license', '')),
+            ('Keywords', u', '.join([u'`{}`'.format(k) for k in assure_list(meta.get('tag', []))])),
+            ('Funding', meta.get('fundedby', '')),
+            ):
+        if label and content:
+            metainfo += u'\n\n### {}\n\n{}'.format(label, content)
+        elif content:
+            metainfo += u'\n\n{}'.format(content)
```

Comment:
apparently no test triggers any of the above lines enhancing metainfo.  Would be nice to have at least a very basic test to see if formatting looks ok

Reply:
Implemented.